zircon.core/src/commonMain/kotlin/org/hexworks/zircon/internal/data/DefaultStackedTile.kt

```diff
@@ -22,13 +23,23 @@ class DefaultStackedTile(
 
     override fun withBaseTile(tile: Tile): StackedTile = DefaultStackedTile(
             baseTile = tile,
-            rest = rest.add(tile)
+            rest = rest
     )
 
-    override fun withRemovedTile(tile: Tile): StackedTile = DefaultStackedTile(
-            baseTile = tile,
-            rest = rest.remove(tile)
-    )
+    override fun withRemovedTile(tile: Tile): StackedTile {
```

Comment:
I think this change overcomplicates the very simple implementation that was present here. In the old implementation it was impossible to remove the `baseTile` and it is clear from the docs that one can't do that. I think we should keep the old implementation here instead:

```kotlin
    override fun withRemovedTile(tile: Tile): StackedTile = DefaultStackedTile(
            baseTile = tile,
            rest = rest.remove(tile)
    )
```